### DISCUSSION TITLE
Quality Reference Image vs. Regcognition Image

### DISCUSSION THREAD

**Discussion by rikmueller** [jomjol/AI-on-the-edge-device#2767] [General]
Hi! 
First of all: What a fantastic project! Thank you jomyol for you effort providing this to us!
I started using it a week ago, and mainly it works well for me.

But I have a question/problem: Why is the picture quality of the recognition image worse than the reference image?
I sometimes get errors when leading "0" are recognized as "8". This happens suddenly after a lot of correct recognitions, and I have no explainition for it. When I look at the recognition image, it looks not very sharp. When I take a new reference image at the same time, it´s crystal clear.

Thanks, Rik

Reference Image:
![Reference Image](https://github.com/jomjol/AI-on-the-edge-device/assets/81533253/3737364c-5266-45aa-bdf2-6399191f9f44)

Recognition Image:
![Recognition Image](https://github.com/jomjol/AI-on-the-edge-device/assets/81533253/23b1929a-26cd-43c1-a806-9a0b12dffe8c)

**Comment by jomjol** [comment]
If you adjust the reference image, then the rotation etc. is done in the brower on the client. This has antialiasing etc. enabled. 
The much simpler image processing procedures on the ESP32 do not have this features due to the limited RAM and processing performance.

But normally this is not a problem for the recognition, as it is trained with images from the ESP32.
